### PR TITLE
PLANNER-1087: Only include shifts/avaliabilities between start and end dates

### PR DESCRIPTION
**Pull request by Christopher-Chianelli**

### CODE REVIEW

File: optashift-employee-rostering-server/src/main/java/org/optaplanner/openshift/employeerostering/server/roster/RosterRestServiceImpl.java [hunk 1]
```diff
@@ -242,17 +255,17 @@ protected EmployeeRosterView getEmployeeRosterView(Integer tenantId, LocalDate s
...
         List<EmployeeAvailability> employeeAvailabilityList = entityManager.createNamedQuery(
-                "EmployeeAvailability.findAll", EmployeeAvailability.class)
+                "EmployeeAvailability.filterWithEmployee", EmployeeAvailability.class)
                 .setParameter("tenantId", tenantId)
+                .setParameter("startDateTime", startDate.atStartOfDay(tenantConfig.getTimeZone()).toOffsetDateTime())
+                .setParameter("endDateTime", endDate.atStartOfDay(tenantConfig.getTimeZone()).toOffsetDateTime())
+                .setParameter("employees", employeeSet)
                 .getResultList();
         for (EmployeeAvailability employeeAvailability : employeeAvailabilityList) {
-            if (employeeSet.contains(employeeAvailability.getEmployee())) {
```
Comment (ge0ffrey) [inline optashift-employee-rostering-server/src/main/java/org/optaplanner/openshift/employeerostering/server/roster/RosterRestServiceImpl.java]: Removing this if looks fishy?

Comment (ge0ffrey) [reply optashift-employee-rostering-server/src/main/java/org/optaplanner/openshift/employeerostering/server/roster/RosterRestServiceImpl.java]: ah nevermind, the queries do an "in :employees" check

File: optashift-employee-rostering-server/src/main/java/org/optaplanner/openshift/employeerostering/server/roster/RosterRestServiceImpl.java [hunk 2]
```diff
@@ -221,19 +228,25 @@ protected EmployeeRosterView getEmployeeRosterView(Integer tenantId, LocalDate s
...
         Set<Employee> employeeSet = new HashSet<>(employeeList);
+        TenantConfiguration tenantConfig = entityManager.createNamedQuery("TenantConfiguration.find",
+                TenantConfiguration.class)
+                .setParameter("tenantId", tenantId).getSingleResult();
+
+        List<Shift> shiftList = entityManager.createNamedQuery("Shift.filterWithEmployees", Shift.class)
+                .setParameter("tenantId", tenantId)
+                .setParameter("startDateTime", startDate.atStartOfDay(tenantConfig.getTimeZone()).toOffsetDateTime())
+                .setParameter("endDateTime", endDate.atStartOfDay(tenantConfig.getTimeZone()).toOffsetDateTime())
+                .setParameter("employees", employeeSet)
```
Comment (ge0ffrey) [inline optashift-employee-rostering-server/src/main/java/org/optaplanner/openshift/employeerostering/server/roster/RosterRestServiceImpl.java]: Can we use the same variable name in the JPA-QL query as we do in the rest code? So rename "employees" to "employeeSet".

**Comment by Christopher-Chianelli** [comment]
@ge0ffrey Still need to find out why our IT framework crashes on addShift (need to be able to add shifts to unit test this)

File: optashift-employee-rostering-server/src/main/java/org/optaplanner/openshift/employeerostering/server/roster/RosterRestServiceImpl.java (hunk 2 above)
Comment (Christopher-Chianelli) [reply optashift-employee-rostering-server/src/main/java/org/optaplanner/openshift/employeerostering/server/roster/RosterRestServiceImpl.java]: Fixed

File: optashift-employee-rostering-server/src/main/java/org/optaplanner/openshift/employeerostering/server/roster/RosterRestServiceImpl.java.orig [hunk 3]
```diff
@@ -0,0 +1,380 @@
+/*
```
Comment (ge0ffrey) [inline optashift-employee-rostering-server/src/main/java/org/optaplanner/openshift/employeerostering/server/roster/RosterRestServiceImpl.java.orig]: I presume this java.orig file didn't need to be committed?

**Pull request merged**